api/app/Http/Controllers/ChildEntityController.php

```diff
@@ -290,82 +319,84 @@ public function deleteOne($id, $childId)
+            $childModels = $this->getManyChildEntitiesByIdsAndParent($ids,$model);
         }
 
+
         if (!empty($childModels) && $childModels->count() !== count($ids)) {
             throw $this->notFoundManyException($ids, $childModels, $childKey);
         }
 
-        /** @var BaseModel[] $childModels */
-        foreach ($childModels as $childModel) {
-            $childModel->markAsDeleted();
-        }
-
-        $model->setRelation($this->relationName, $childModels, $relation);
-        $this->saveModel($model);
+        ModelHelper::deleteMany($childModels);
 
         return $this->getResponse()->noContent();
     }
 
     /**
-     * @param $id
      * @return BaseModel
      */
-    protected function getParentModel($id)
+    public function getParentModel()
     {
-        $this->validateId($id, $this->getKeyName(), $this->validateRequestRule);
+        return $this->parentModel;
+    }
 
-        try {
-            return $this->getRepository()->find($id);
-        } catch (ModelNotFoundException $e) {
-            throw $this->notFoundException($this->getKeyName());
-        }
+    /**
+     * @return ChildBaseModel
+     */
+    public function getChildModel()
+    {
+        return $this->childModel;
     }
 
     /**
-     * @param BaseModel $model
-     * @throws \Exception
-     * @throws \Exception|null
-     * @throws \Spira\Repository\Repository\RepositoryException
+     * @param $id
+     * @return BaseModel
      */
-    protected function saveModel(BaseModel $model)
+    protected function getParentEntityById($id)
     {
-        try {
-            $this->getRepository()->save($model);
-        } catch (ValidationException $e) {
-            if ($exception = $model->getRelationErrors($this->relationName)) {
-                throw $exception;
-            }
+        return $this->getParentModel()->findOrFail($id);
+    }
 
-            throw $e;
-        }
+    /**
+     * @param $id
+     * @param $parent
+     * @return BaseModel
+     */
+    protected function getChildEntityByIdAndParent($id, $parent)
+    {
+        return $this->getChildModel()->findByIdAndParent($id,$parent);
     }
 
     /**
-     * @param BaseModel $model
-     * @return HasOneOrMany
+     * @param $ids
+     * @param $parent
+     * @return Collection
      */
-    protected function getRelation(BaseModel $model)
+    protected function getManyChildEntitiesByIdsAndParent($ids, $parent)
     {
-        return $model->{$this->relationName}();
+        return $this->getChildModel()->findManyByIdsAndParent($ids,$parent);
     }
 
     /**
-     * @param Relation $relation
-     * @return string
+     * @param $parent
+     * @return Collection
      */
-    protected function getChildKey(Relation $relation)
+    protected function getAllChildEntitiesByParent($parent)
     {
-        return $relation->getQuery()->getModel()->getKeyName();
+        return $this->getChildModel()->findAllByParent($parent);
     }
+
```

Comment:
this way we can't check some complex rules, like createOnly or unique
+ we leaving it inside a model is a great way of changing business rule without much fuss.
ex. Me have an article and text is not required, we decide to make it required, we just change validation rule and on the next patch request user will know, that he needs to add text

the thing is, that we can really make use of visitor pattern in controller with
`$this->validate($model)`